persistence/block.go

```diff
@@ -12,6 +12,7 @@ import (
 	coreTypes "github.com/pokt-network/pocket/shared/core/types"
 )
 
+// INCOMPLETE: Need to think through how we will handle pruned vs non-pruned txIndexers
```

Comment:
I think that there's a lot packed into this comment, shall we create an issue to describe what you mean in detail and tag it here?